boards/opentitan/src/tests/spi_host.rs

```diff
@@ -0,0 +1,232 @@
+use crate::tests::run_kernel_op;
+use crate::PERIPHERALS;
+use core::cell::Cell;
+#[allow(unused_imports)]
+use kernel::hil::spi::{ClockPhase, ClockPolarity};
+use kernel::hil::spi::{SpiMaster, SpiMasterClient};
+use kernel::static_init;
+use kernel::utilities::cells::TakeCell;
+use kernel::{debug, ErrorCode};
+
+struct SpiHostCallback {
+    transfer_done: Cell<bool>,
+    tx_len: Cell<usize>,
+    tx_data: TakeCell<'static, [u8]>,
+    rx_data: TakeCell<'static, [u8]>,
+}
+
+impl<'a> SpiHostCallback {
+    fn new(tx_data: &'static mut [u8], rx_data: &'static mut [u8]) -> Self {
+        SpiHostCallback {
+            transfer_done: Cell::new(false),
+            tx_len: Cell::new(0),
+            tx_data: TakeCell::new(tx_data),
+            rx_data: TakeCell::new(rx_data),
+        }
+    }
+
+    fn reset(&self) {
+        self.transfer_done.set(false);
+        self.tx_len.set(0);
+    }
+}
+
+impl<'a> SpiMasterClient for SpiHostCallback {
+    fn read_write_done(
+        &self,
+        tx_data: &'static mut [u8],
+        rx_done: Option<&'static mut [u8]>,
+        tx_len: usize,
+        _rc: Result<(), ErrorCode>,
+    ) {
+        //Transfer Complete
+        assert_eq!(tx_len, self.tx_len.get());
+
+        //Capture Buffers
+        self.tx_data.replace(tx_data);
+        if rx_done.is_none() == false {
```

Comment:
You don't need the `== false`.

You should be able to use an `if let Some()..` here which is more Rusty